test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildADesktopExe.cs

```diff
@@ -0,0 +1,113 @@
+                .CopyTestAsset("DesktopMinusRid")
+                .WithSource()
+                .Restore();
+
+            var buildCommand = new BuildCommand(Stage0MSBuild, testAsset.TestRoot);
+            buildCommand
+                .MSBuild
+                .CreateCommandForTarget("build", buildCommand.FullPathProjectFile)
+                .CaptureStdOut()
+                .Execute()
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining("RuntimeIdentifier must be set");
+        }
+
+        [Theory]
+        [InlineData("win7-x86", "x86")]
+        [InlineData("win8-x86-aot", "x86")]
+        [InlineData("win7-x64", "x64")]
+        [InlineData("win8-x64-aot", "x64")]
+        [InlineData("win10-arm", "arm")]
+        [InlineData("win10-arm-aot", "arm")]
+        //PlatformTarget=arm64 is not supported and never inferred
+        [InlineData("win10-arm64", "AnyCPU")]
+        [InlineData("win10-arm64-aot", "AnyCPU")]
+        // cpu architecture is never expected at the front
+        [InlineData("x86-something", "AnyCPU")]
+        [InlineData("x64-something", "AnyCPU")]
+        [InlineData("arm-something", "AnyCPU")]
+        public void It_builds_with_inferred_platform_target(string runtimeIdentifier, string expectedPlatformTarget)
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return;
+            }
+
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("DesktopMinusRid", identifier: Path.DirectorySeparatorChar + runtimeIdentifier)
+                .WithSource()
+                .Restore("", $"/p:RuntimeIdentifier={runtimeIdentifier}");
+
+            var buildCommand = new BuildCommand(Stage0MSBuild, testAsset.TestRoot);
+            buildCommand
+                .Execute($"/p:RuntimeIdentifier={runtimeIdentifier}")
+                .Should()
+                .Pass();
+
+            var assemblyPath = Path.Combine(buildCommand.GetOutputDirectory("net46").FullName, "DesktopMinusRid.exe");
+            var assemblyInfo = AssemblyInfo.Get(assemblyPath);
+            assemblyInfo
+                .Should()
+                .Contain(
+                    "AssemblyDescriptionAttribute",
+                    $"PlatformTarget={expectedPlatformTarget}");
+        }
+
+        [Fact]
+        public void It_respects_explicit_platform_target()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return;
+            }
+
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("DesktopMinusRid")
+                .WithSource()
+                .Restore("", $"/p:RuntimeIdentifier=win7-x86");
+
+            var buildCommand = new BuildCommand(Stage0MSBuild, testAsset.TestRoot);
+            buildCommand
+                .Execute($"/p:RuntimeIdentifier=win7-x86", "/p:PlatformTarget=x64")
```

Comment:
Is there ever a case where it would be correct to do this (ie have `RuntimeIdentifier` be for x86 but `PlatformTarget` be x64)?